File: xamoom-pingeborg-ios/xamoom-pingeborg-ios/NotificationHelper.m
//
//  NotificationHelper.m
//  pingeb.org
//
//  Created by Raphael Seher on 20/06/2017.
//  Copyright © 2017 xamoom GmbH. All rights reserved.
//

#import <UserNotifications/UserNotifications.h>
#import "NotificationHelper.h"

@implementation NotificationHelper

+ (void)removeAllNotifications {
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  [center removeAllDeliveredNotifications];
  [center removeAllPendingNotificationRequests];
  
  [[UIApplication sharedApplication] cancelAllLocalNotifications];
  [[UIApplication sharedApplication] setApplicationIconBadgeNumber:0];
  [[UIApplication sharedApplication] cancelAllLocalNotifications];
}

+ (void)showNotificationWithTitle:(NSString *)title andBody:(NSString *)body {
  UILocalNotification *notification = [[UILocalNotification alloc] init];
  [notification setAlertTitle:title];
  [notification setAlertBody:body];
  [notification setFireDate:[[NSDate alloc] init]];
  
  [[UIApplication sharedApplication] scheduleLocalNotification:notification];
}

@end

File: xamoom-pingeborg-ios/xamoom-pingeborg-ios/NotificationHelper.swift
//
//  NotificationHelper.swift
//  pingeb.org
//
//  Created by Raphael Seher on 20/06/2017.
//  Copyright © 2017 xamoom GmbH. All rights reserved.
//

import UIKit
import UserNotifications

class NotificationHelper {
  
  static func removeAllNotifications() {
    if #available(iOS 10.0, *) {
      let center = UNUserNotificationCenter.current()
      center.removeAllPendingNotificationRequests()
      center.removeAllDeliveredNotifications()
    } else {
      UIApplication.shared.cancelAllLocalNotifications()
    }
    UIApplication.shared.applicationIconBadgeNumber = 0
    UIApplication.shared.cancelAllLocalNotifications()
  }
  
  static func showNotification(title: String, body: String) {
    let notification = UILocalNotification()
    notification.alertTitle = title
    notification.alertBody = body
    notification.fireDate = Date()
    
    UIApplication.shared.scheduleLocalNotification(notification)
  }
}
